.github/workflows/docs-deployment.yaml

```diff
@@ -0,0 +1,98 @@
+
+jobs:
+  package-push-docs:
+    strategy:
+      # Do not fail-fast because we need to ensure all tests go to completion
+      # so we try not to get hanging machines
+      fail-fast: false
+      matrix:
+        arch: [grayskull]
+    env:
+      TT_METAL_ENV: ${{ vars.TT_METAL_ENV }}
+      DOCS_VERSION: v0.50.0
+      ARCH_NAME: ${{ matrix.arch }}
+      LOGURU_LEVEL: INFO
+      LD_LIBRARY_PATH: ${{ github.workspace }}/build/lib
+    environment: dev
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          lfs: false
+      - name: Set up dynamic env vars for build
+        run: |
+          echo "TT_METAL_HOME=$(pwd)" >> $GITHUB_ENV
+      - uses: ./.github/actions/install-metal-deps
+        with:
+          os: ubuntu-20.04
+      - uses: ./.github/actions/install-metal-dev-deps
+        with:
+          os: ubuntu-20.04
+      - uses: actions/download-artifact@v4
+        with:
+          name: TTMetal_build_${{ matrix.arch }}
+      - name: Extract files
+        run: tar -xvf ttm_${{ matrix.arch }}.tar
+      - uses: ./.github/actions/install-python-deps
+      - name: Build Doxygen Docs
+        timeout-minutes: 15
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: |
+          source ${{ github.workspace }}/python_env/bin/activate
+          cd $TT_METAL_HOME
+          export PYTHONPATH=$TT_METAL_HOME
+          doxygen
+      - name: Clone Shared Docs Repo
+        run: |
+          GITHUB_USER=tenstorrent-github-bot
+          git config --global user.email "${GITHUB_USER}@tenstorrent.com"
+          git config --global user.name "${GITHUB_USER}"
+          echo "About to clone"
+          git clone https://${GITHUB_USER}:${{ secrets.TTBOT_DOCS_ACCESS }}@github.com/tenstorrent/docs-test.git
+          git remote set-url origin https://${GITHUB_USER}::${{ secrets.TTBOT_DOCS_ACCESS }}@github.com/tenstorrent/docs-test.git
+      - name: Package TT-Metalium Docs
+        run: |
+          echo "Preparing tt-metalium docs assets"
+          mkdir tt-metalium
+          cp -Lr docs/doxygen_build tt-metalium
+          cp -Lr docs/source/tt-metalium/* tt-metalium
+          rm tt-metalium/conf.py  # Do not change conf.py
+          echo "Copying the docs into the repo"
+          cp -r tt-metalium/* docs-test/tt-metalium
+      - name: Package TT-NN Docs
+        run: |
+          mkdir ttnn-docs
+          cp -Lr docs/doxygen_build ttnn-docs
+          cp -Lr docs/source/ttnn/* ttnn-docs
+          rm ttnn-docs/conf.py  # Do not change conf.py
+          echo "Copying the docs into the repo"
+          cp -r ttnn-docs/* docs-test/ttnn
+      - name: Push new versions to the Docs repo for publishing
+        run: |
+          cd docs-test
+          echo "Register new documentation versions"
+          python update_tags.py tt-metalium $DOCS_VERSION
+          python update_tags.py ttnn $DOCS_VERSION
+
+          git add .
```

Comment:
Hmm, so we did decide to build on the docs repo?

Reply:
Yeap. The issue is that if we build here and convert to .md some data would be missing. 
The alternative would be to build to .rst and then commit that into the repo. 
Let's explore if we could make it happen in the follow up issue (I'll create one ).

Reply:
https://github.com/tenstorrent/tt-metal/issues/10450